### PR TITLE
CA-205586: Reserve orginal IP settings before set IP by agent

### PR DESCRIPTION
**Pull request by liulinC**
Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by benchalmers** [comment]
I don't think that this is sufficient: CLM is - I presume - quite likely to want to be able to reboot the VM while the static ip address is set.  As far as I can see, this code is only storing the network details in memory - I think they need to be stored in the registry

**Comment by liulinC** [comment]
Hi ben, 
 
Actually, The static IP address information is stored on the disk, the file path is:
private static string iniFile = System.IO.Path.GetDirectoryName(Application.ExecutablePath) +  "\\ipsetting.ini";
It seems storing the message on the disk can work as well even the machine is restart. 

If I misunderstand anything, please let me know.
Regards!
Lin

**Comment by benchalmers** [comment]
Oh yes - you're right - it is writing to an ini file.

I think I would still prefer to see this persisted in the registry - there is a good article here (only 9 years old) about why the use of .ini files was deprecated in favour of using the registry for config info

https://blogs.msdn.microsoft.com/oldnewthing/20071126-00/?p=24383

**Comment by liulinC** [comment]
@benchalmers, Thanks for your comments, 
I also prefer your suggestion which store the ip information in the registry. 
The coding is done and a new pull request is submitted,
Could you help to review the code? Thanks and Regards!

File: src/xenguestlib/FeatureStaticIpSetting.cs [hunk 1]
```diff
@@ -41,6 +41,152 @@
...
+            get { return _ip; }
+        }
+        private string _mask;
+        public string MASK
+        {
+            set { _mask = value; }
+            get { return _mask; }
+        }
+        private string _gateway;
+        public string GATEWAT
```
Comment (benchalmers) [inline src/xenguestlib/FeatureStaticIpSetting.cs]: I think this should be GATEWAY - not GATEWAT

Comment (liulinC) [inline src/xenguestlib/FeatureStaticIpSetting.cs]: @benchalmers, thanks for  your help, 
The wording is fixed and the merge ticked is created, ticket number CA-205789.
Could you help to merge it.

**Pull request merged**